examples/example.rs

```diff
@@ -43,7 +45,9 @@ fn main() {
     }
 
     // middleware is optional and can be registered with `utilize`
-    server.utilize(logger);
+    // issue #15444
+    let logger_handler: fn(&Request, &mut Response) -> MiddlewareResult = logger;
+    server.utilize(logger_handler);
```

Comment:
I think it's worth trying:
`type HandlerPtr = fn(&Request, &mut Response) -> MiddlewareResult`
and changing fallout code to:
`foo(bar as HandlerPtr)`

*OR* I think changing the naming convention, `logger_handler` => `logger_fnptr`.

I guess it's a nit really, but I'm guessing people may understand a cast to a function pointer if they come from c++/c rather than reading 'I'm turning my handler into a handler..?' and perhaps being confused.

cc @SimonPersson